llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp

```diff
@@ -3593,6 +3593,78 @@ static Value *foldOrOfInversions(BinaryOperator &I,
   return nullptr;
 }
 
+// A decomposition of ((A & N) ? 0 : N * C) . Where X = A, Factor = C, Mask = N.
```

Comment:
I think it would be clearer if you wrote this directly with the correct names instead of specifying the mapping separately: `(X & Mask) ? 0 : Mask * Factor`